### DISCUSSION TITLE
capillary pressure -another try

### DISCUSSION THREAD

**Discussion by marinsiebert** [idaholab/moose#29379] [Q&A Modules: Porous Flow]
Hello everyone, 

this is a continuation of discussion #26790, so I guess one you guys @GiudGiud @cpgr @WilkAndy would be the ones to adress. So far I've been through the user manual and the available objects multiple times, however with no success. Maybe you can clarify this:

The sensitivity I am talking about in the discussion above does not seem to be related to wrong calculations in the capillary pressure as the approximation by van genuchten was chosen to fit available WRC-data.

Since the capillary pressures are fine this must be related to either 
**a) insufficiently reproduced mass transport:** 
in unsaturated porous media mass transport is often related to isothermal and thermal vapor  and liquid transport, such as described here in section 2.3.: [https://acsess.onlinelibrary.wiley.com/doi/full/10.1002/vzj2.20173](url)
This results in a equilibrium where water flows towards an area of increased temperature and vapor is transported away, up to a point where this equilibrium holds no longer such that no liquid flow occurs, leading to a dry area. 

I tried going a little bit in that direction using the ThermoDiffusion Kernel, which actually improved the results a little bit, however still not really satisfying:

![claypsi](https://github.com/user-attachments/assets/15543319-84f5-4187-8bb7-1937776f1605) showing a lack of sensitivity but nicely repocing the shape
![sandpsi](https://github.com/user-attachments/assets/4f450b4e-eb4c-4e1d-8229-d207dd1a239d) showing kind of an inversed behaviour compared to the measured data (most likely due to drying front now progressing as fast) and also lack of sensitivity

**b) insufficiently reproduced phase change (evaporation):**
assuming that the capillary pressures are calculated correctly and mass transport is fine (more or less? see a)), maybe the amount of water evaporation due to increased temperature as calculated by IAWPS97 does not work properly (doubt that) or is simply not entirely applicable to unsaturated porous media? 

Right now I tend to think, that option a) is the likely candidate and that the mass transport equations are simply not appropiate for this type of problem. 

I would be very happy to hear your opinions on this. 

marin

**Comment by marinsiebert** [comment]
It has been real quiet on this end and unfortunately I have no spare time left, so I will close the discussion as outdated. 

Some closing remarks on this if someone else tries to do something simila in the future: 
The problem most likely lies in the fact, that thermodiffusion is not incorporated in the standard tool-box of porous flow. This can be done either by using a ThermoDiffusion or SoretDiffusion Kernel. 
Alternatively you can also populate a GenericMaterial with your own description of the thermal diffusion coefficients of your phases and then pass that to MatDiffusion, acting on your pressure, saturation or mass fraction (depending on the model you are using) and specifying you temperature variable as gradient.

**Reply by GiudGiud** [reply]
Sorry I'm not actually a porous flow person so I could not help. You have the right experts tagged, they are likely just busy

**Reply by cpgr** [reply]
Did you manage to get this working properly @marinsiebert? 

Sorry I missed it last year but my end of year was hectic!

**Reply by marinsiebert** [reply]
Unfortunately my contract ended and so has my time in research, so I was did not have the opportunity to delve in deeper, however, I am confident that the approach described above could work. I did a short simulation with the MatDiffusion Kernel together with the GenericMaterial, where the results were very good for the first heating, however, for subsequent heating phases the pressures show progressively increasing overshoot. 
The diffusion coefficient in this case were based on De Vries (1958), so I guess they no not go well with the mass diffusion coefficients in my model. They may also not be applicable to this case with the generalised Darcy's law (As in the one PorousFlow uses if I am not mistaken), since De Vries does not explicitly describe the flow of the gaseous phase and uses Richards equation. . 

This (MOYNE, C.; PERRE, P.. (1991). PROCESSES RELATED TO DRYING: PART I, THEORETICAL MODEL. , 9(5), 1135–1152. doi:10.1080/07373939108916746 ) might work, though. 

Anyhow, I think the module in the current state needs some tweaking to be applicable for the vadose zone (which makes sense, because I think it was initially developed for rock?)
